packages/core/src/codewhisperer/activation.ts

```diff
@@ -286,7 +291,46 @@ export async function activate(context: ExtContext): Promise<void> {
                     await getChatAuthState()
                 ).codewhispererChat
             )
+            // when changing connection state in Q, also change connection state in toolkit
```

Comment:
we need a way to register these callbacks if the toolkit is installed after Q. As discussed, we should put these registrations behind a command and a `once()`. Q can call its own command on startup if toolkit is already installed. Toolkit can `tryExecute()` this command on its activation, covering the case if it is installed later on.
I think this would also eliminate deadlock on line 275, and we can change line 274 to be `isExtensionActive()` instead.

Reply:
This is fixed in latest commit